.github/actions/average_single_benchmark/action.yml

```diff
@@ -0,0 +1,68 @@
+name: Average of a single benchmark
+description: runs individual benchmark based on inputs provided
+
+inputs:
+  path:
+    required: true
+    description: e.g. simeng-benchmarks/binaries/CloverLeaf/openmp
+    
+  name:
+    required: true
+    description: e.g. cloverleaf_gcc10.3.0_armv8.4+sve
+
+  datafile_path:
+    required: true
+    description: The path to the corresponding data file for that benchmark e.g. simeng-benchmarks/Data_Files/CloverLeaf.
+
+  runs:
+    required: true
+    default: 10
+    description: The number of times a single benchmark should be averaged over.
+
+runs:
+  using: 'composite'
+  steps:
+
+    ############################################################################
+    # run individual benchmark + calculate and store into env variable via echo
+    ############################################################################
+    - name: run benchmark & compare runtimes
+      shell: bash
+      run: |
+          benchmark_path=${{ inputs.path }}
+          benchmark_name=${{ inputs.name }}
+          datafile_path=${{ inputs.datafile_path }}
+
+          output_file="benchmark_output.txt"
+          total_time=0
+          runs=${{ inputs.runs }}
+
+          cd "${{ github.workspace }}/$benchmark_path"
+
+          # Loop to run the benchmark 'runs' times
+          for (( i=1; i<=runs; i++ ))
+          do  
+              # Run the benchmark and redirect output to a file
+              if [ $datafile_path ]; then
+                  sudo simeng "${{ github.workspace }}/configs/a64fx.yaml" $benchmark_name -n 64 -i 1 --deck "${{ github.workspace }}/$datafile_path"  > "$output_file"
+              else
+                  sudo simeng "${{ github.workspace }}/configs/a64fx.yaml" $benchmark_name  > "$output_file"
+              fi
```

Comment:
I think this could be generalised more. Instead of checking whether a "data file path" has been provided and hard coding the rest of the miniBUDE input, the "function" should have a `workload_parameters` input. This can be empty or a string of many parameters. 

This "function" would then simply call `sudo simeng a64fx $workload_parameters` and would be completely generalised to the workload. This would be more easily understandable in my opinion.

Not sure the best way to achieve this currently

Reply:
Having found the correct branch on the simeng-benchmarks repo, you could somehow pipe the `cmd` field from the json file directly here

Reply:
This is actually superfluous for the non-performance workflows.  Have removed from the PR.